projects/cork-v1/index.js

```diff
@@ -0,0 +1,123 @@
+     * - Fetches logs from the factory contract to identify relevant tokens.
+     * - Extracts unique external token addresses from logs.
+     * - Uses sumTokens2 to aggregate TVL data.
+     *
+     * @param {Object} api - API object providing blockchain access.
+     * @returns {Promise<Object>} Total value locked (TVL) balances.
+     */
+    tvl: async (api) => {
+      // Redemption Assets (RA) in Cork AMM pools
+      const logs3 = await getLogs2({ api, factory: ammHook, eventAbi: eventAbi3, fromBlock, skipCache: SKIP_CACHE, })
+      const tokens = logs3.map(({ ra }) => ra) // RA
+      const calls = logs3.map(({ ra, ct }) => ({
+        target: ammHook,
+        params: [ra, ct],
+      }))
+      const bals = await api.multiCall({
+        abi: 'function getReserves(address token0, address token1) external view returns (uint256 reserve0, uint256 reserve1)',
+        calls,
+      })
+      api.addTokens(tokens, bals.map(({ reserve0 }) => reserve0))
+
+      // Redemption Assets (RA) and Pegged Assets (PA) in Cork PSM pools
+      const logs1 = await getLogs2({ api, factory: poolManager, eventAbi: eventAbi1, fromBlock, skipCache: SKIP_CACHE, })
+      const tokenSet = new Set()
+      const ownerTokens = []
+      logs1.forEach(log => {
+        tokenSet.add(log.pa) // RA
+        tokenSet.add(log.ra) // PA
+        // if (log.lv !== nullAddress) {
+        //   ownerTokens.push([[log.pa, log.ra], log.lv])
+        // }
+      })
+      ownerTokens.push([Array.from(tokenSet), poolManager])
+      return sumTokens2({ api, ownerTokens, permitFailure: true, })
+    },
+
+    /**
+    * Calculates LP tokens (where one side of the market is the platform's own token)
+    * locked or staked in the platform's contracts.
+    * - Fetches logs for liquidity pools.
+    * - Extracts LP tokens from logs.
+    * - Uses sumTokens2 to aggregate the Pool2 balances.
+    *
+    * @param {Object} api - API object providing blockchain access.
+    * @returns {Promise<Object>} Pool2 balances.
+    */
+    pool2: async (api) => {
+      // Only LP tokens in Cork Vault pools
+      const logs3 = await getLogs2({ api, factory: ammHook, eventAbi: eventAbi3, fromBlock, skipCache: SKIP_CACHE, })
+      const lpTokens = []
+      logs3.forEach(log => lpTokens.push(log.liquidityToken)) // LPT
+      const stakingContracts = [poolManager]
+      return sumTokens2({ api, tokens: lpTokens, owners: stakingContracts, resolveLP: false, })
+    },
+
+    /**
+     * Calculates the platform's own tokens locked or staked
+     * in the platform's contracts.
+     * - Fetches staking-related logs from core and hook contracts.
+     * - Extracts the platform's own tokens from logs.
+     * - Uses sumTokens2 to aggregate the Staking balances.
+     *
+     * @param {Object} api - API object providing blockchain access.
+     * @returns {Promise<Object>} Staking balances.
+     */
+    staking: async (api) => {
```

Comment:
please remove this staking function, it seems like this is double counting TVL of tvl()